### PR TITLE
7871 delete map plugin

### PR DESCRIPTION
**Pull request by ale-cristofori**
## Description
This PR contains the code developing the `DeleteMap` plugin. This plugin, when configured and activated allows authenticated and authorised users to delete the current map from the map itself. The plugin sits as a new item in the BurgerMenu plugin.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

#7871

**What is the current behavior?**
The only way authenticated and authorised user can delete a saved map is from the cards grid in the home page. Users can click on the delete button on every single resource card item.

**What is the new behavior?**
 A new entry in the burger menu, the `DeleteMap` plugin is added to the actions list. Users can perform map deletion from the current map, rather than searching the map they want to delete in the home page. A confirm modal window appears and if users confirm map deletion they are redirected to the home page, a confirm deletion banner will appear.

recording_1

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by tdipisa — CHANGES_REQUESTED** [review]
@ale-cristofori Thank you for this. There are some test failures to fix to proceed with the review.

**Comment by tdipisa** [comment]
@ale-cristofori I'm sorry but since this PR is still in pending review (for a while I know), there are some conflicts to solve now. Can you please do it? Thank you.

**Comment by ale-cristofori** [comment]
> @ale-cristofori I'm sorry but since this PR is still in pending review (for a while I know), there are some conflicts to solve now. Can you please do it? Thank you.

@tdipisa Yes, I am on it

File: web/client/plugins/DeleteMap.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+}
+
+DeleteConfirmDialog.propTypes = {
+    show: PropTypes.bool,
+    mapId: PropTypes.string,
+    onConfirmDelete: PropTypes.func,
+    onClose: PropTypes.func
+};
+
```
Comment (MV88) [inline web/client/plugins/DeleteMap.jsx:61]: let's stick to one way of doing things this way of declaring PropTypes is mostly used for functional components
if you use classes let's move this inside the class component

File: web/client/plugins/DeleteMap.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+DeleteConfirmDialog.contextTypes = {
```
Comment (MV88) [inline web/client/plugins/DeleteMap.jsx:62]: this seems a duplication

File: web/client/plugins/DeleteMap.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    router: PropTypes.object
+};
+
+const showConfirmDeleteMapModalSelector = state => state.controls && state.controls.mapDelete && state.controls.mapDelete.enabled;
```
Comment (MV88) [inline web/client/plugins/DeleteMap.jsx:66]: I know you are testing this in the plugin, 
but can i ask you also to add a unit test just for this selector and 
to move it in the controls selector ?

**Review by MV88 — CHANGES_REQUESTED** [review]
just a few changes here and there

File: web/client/plugins/DeleteMap.jsx (hunk 2 above)
Comment (ale-cristofori) [reply web/client/plugins/DeleteMap.jsx]: Resolved

File: web/client/plugins/DeleteMap.jsx (hunk 1 above)
Comment (ale-cristofori) [reply web/client/plugins/DeleteMap.jsx]: Reolved

File: web/client/plugins/DeleteMap.jsx (hunk 3 above)
Comment (ale-cristofori) [reply web/client/plugins/DeleteMap.jsx]: Resolved

File: web/client/plugins/DeleteMap.jsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            >
+                <div className="ms-detail-body">
+                    <Message msgId="resources.deleteConfirmMessage" />
+                </div>
+            </ConfirmDialog>
+        );
+    }
+}
+
+export const showConfirmDeleteMapModalSelector = state => state.controls && state.controls.mapDelete && state.controls.mapDelete.enabled;
```
Comment (MV88) [inline web/client/plugins/DeleteMap.jsx:62]: @ale-cristofori i meant to move this to the [controls selector](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/selectors/controls.js) file an import it here

and add a unit test [here](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/selectors/__tests__/controls-test.js)

**Review by MV88 — CHANGES_REQUESTED** [review]
@ale-cristofori 
please check my latest comments?

also may I ask you why you merge master over your local branches?
is there any particular reasons?

**Review by MV88 — CHANGES_REQUESTED** [review]

**Review by MV88 — APPROVED** [review]

**Comment by ElenaGallo** [comment]
Test passed, @ale-cristofori please backport to 2022.01.xx. Thanks

**Pull request merged**